source/expressions.tex

```diff
@@ -1636,18 +1624,43 @@
 The closure type for a \grammarterm{lambda-expression} has a public
 inline function call operator (for a non-generic lambda) or
 function call operator template (for a generic lambda)\iref{over.call}
-whose parameters and return type
-are described by the \grammarterm{lambda-expression}'s
-\grammarterm{parameter-declaration-clause} and \grammarterm{trailing-return-type}
-respectively, and whose
-\grammarterm{template-parameter-list} consists of
-the specified \grammarterm{template-parameter-list}, if any.
-The \grammarterm{requires-clause} of the function call operator template
+whose:
+\begin{itemize}
+\item \grammarterm{parameter-declaration-clause} is the \grammarterm{lambda-declarator}'s
+\grammarterm{parameter-declaration-clause},
+
+\item return type is \tcode{auto},
+
+\item \grammarterm{trailing-return-type} is the \grammarterm{lambda-declarator}'s
+\grammarterm{trailing-return-type}, if any,
+
+\begin{note}
+The actual return type will be the type specified by the
+\grammarterm{trailing-return-type} if provided and/or deduced from
+\tcode{return} statements as described in~\ref{dcl.spec.auto}.
+\begin{example}
+\begin{codeblock}
+auto x1 = [](int i){ return i; };       // OK: return type is \tcode{int}
+auto x2 = []{ return { 1, 2 }; };       // error: deducing return type from \grammarterm{braced-init-list}
+int j;
+auto x3 = []()->auto&& { return j; };   // OK: return type is \tcode{int\&}
+\end{codeblock}
+\end{example}
+\end{note}
```

Comment:
Sorry for the piecemeal feedback, but we shouldn't have full-sentence (long) notes inside a normative sentence (which, in this case, spans multiple bullets) if we can help it. Please move the note to after the bulleted list.

Reply:
The most difficult decision: Example after or inside the Note

Reply:
The example is explaining the note, so goes inside the note.